qcodes/instrument_drivers/oxford/IPS120.py

```diff
@@ -707,13 +718,11 @@ def heater_on(self):
         """Switch the heater on, with PSU = Magnet current check"""
         current_in_magnet = self.persistent_current()
         current_in_leads = self.current()
-        if self.switch_heater() == 1:
```

Comment:
i'm confused of how this could even ever work. `switch_heater` parameter, when called `get` on, returns a "string". Before this PR, that string was compared against an integer value of `1` - which is incorrect. Now, thanks to the change in this PR, the "string" coming from `switch_heater()` call gets compared against another string that now comes from `_GET_STATUS_SWITCH_HEATER` (for "index" `1`) - which is correct. Am i mistaken somewhere or just `heater_on` was never used by anyone in this world?

Reply:
You are correct, it never worked.

Reply:
awesome! thanks to refactoring, it all works now :)